core/src/replay_stage.rs

```diff
@@ -383,6 +386,40 @@ impl ReplayStage {
         Ok(())
     }
 
+    fn update_confidence_cache(
+        bank_forks: &Arc<RwLock<BankForks>>,
+        tower: &Tower,
+        lockouts: HashMap<u64, StakeLockout>,
+    ) -> thread::JoinHandle<()> {
+        let total_epoch_stakes = tower.total_epoch_stakes();
+        let mut w_bank_forks = bank_forks.write().unwrap();
+        for (fork, stake_lockout) in lockouts.iter() {
+            if tower.root().is_none() || *fork >= tower.root().unwrap() {
+                w_bank_forks.cache_fork_confidence(
+                    *fork,
+                    stake_lockout.stake(),
+                    total_epoch_stakes,
+                    stake_lockout.lockout(),
+                );
+            }
+        }
+        drop(w_bank_forks);
+        let bank_forks_clone = bank_forks.clone();
+        let root = tower.root();
+        thread::spawn(move || {
```

Comment:
This call to update_confidence_cache happens pretty often. Instead of thread startup/teardown every time, we could keep a long running thread just to handle these computations, we just need to pass it the `lockouts`

Reply:
How does this look?